### PR TITLE
[MINOR][SQL][DOCS] Fix typos in scaladoc strings of higher order functions

### PR DESCRIPTION
**Pull request by zero323**
### What changes were proposed in this pull request?

Fix following typos:

- tranformation -> transformation
- the boolean -> the Boolean
- signle -> single

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

Scala linter.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 1]
```diff
@@ -3457,7 +3457,7 @@ object functions {
    * }}}
    *
    * @param column: the input array column
-   * @param f: col => predicate, the boolean predicate to filter the input column
+   * @param f: col => predicate, the Boolean predicate to filter the input column
```
Comment (srowen) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: This is fine, though 'boolean' is very commonly used as the noun, despite it being from a name

**Review by srowen — APPROVED** [review]
This is OK, but the more you can find and fix at once, the fewer PRs there are to review. If you can search for more while at it, that's helpful.

**Comment by srowen** [comment]
Merged to master

**Comment by zero323** [comment]
Thanks @srowen 

> If you can search for more while at it, that's helpful.

I actually reviewed the whole `functions`, but this was the only hotspot.

**Pull request closed without merging**